Yosemite/Yosemite/Stores/OrderStore.swift

```diff
@@ -384,6 +385,26 @@ private extension OrderStore {
         }
     }
 
+    /// Optimistic updates the specified fields from an order.
+    ///
+    /// Updates will be reverted in case of failure.
+    ///
+    func optimisticUpdateOrder(siteID: Int64, order: Order, fields: [OrderUpdateField], onCompletion: @escaping (Result<Order, Error>) -> Void) {
+        // Optimistically update the stored order.
+        let backupOrder = updateStoredOrder(readOnlyOrder: order)
+
+        remote.updateOrder(from: siteID, order: order, fields: fields) { [weak self] result in
+            guard case .failure = result else {
+                onCompletion(.success(order))
+                return
+            }
+
+            // Revert optimistic update.
+            self?.updateStoredOrder(readOnlyOrder: backupOrder)
```

Comment:
Mm...good point 🤔 You're right. 

In addition, I think there is another problem. As you said, if the order to be updated doesn't exist locally, I'm returning it as the `backupOrder` but this order is not saved in the storage. So, if the request fails, the order will not be saved because the order doesn't exist. 

I think the correct flow should be:

<img width="350px" src="https://user-images.githubusercontent.com/1409041/164386081-dfe140d7-b9f7-47c8-9221-d6488f4b1442.jpg"/>

WDYT? 🤔

Reply:
Inserting the missing order into the local storage sounds good to me! Please make sure to add a `DDLogWarn` in that case to note that something weird is happening so we're saving the order as a workaround.

We should still be able to revert that change by removing the newly inserted item though - how does that sound?

Reply:
> Please make sure to add a DDLogWarn in that case to note that something weird is happening so we're saving the order as a workaround.

Great! 👍🏼 

> We should still be able to revert that change by removing the newly inserted item though - how does that sound?

I really like the idea 👌🏼 I'll create a helper method to `deleteOrRevert` upon failure. Thanks for the suggestion 🤙